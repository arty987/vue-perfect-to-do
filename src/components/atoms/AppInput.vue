<script>
export default {
  name: "AppInput",
  data() {
    return {
      inputValue: "",
    };
  },
  methods: {
    addTask() {
      const inputValue = this.inputValue.trim();
      if (inputValue) {
        this.$store.dispatch("addTask", inputValue);
        this.inputValue = "";
      }
    },
    clearInputValue() {
      this.inputValue = "";
    }
  }
}
</script>

<template>
  <input
      type="text"
      placeholder="Please type your task here..."
      v-model="inputValue"
      maxlength="150"
      @keyup.enter="addTask"
      @keyup.esc="clearInputValue"
  />
</template>

<style scoped>
input {
  width: 100%;
  margin-left: 10px;
  outline: none;
  border: none;
}
</style>