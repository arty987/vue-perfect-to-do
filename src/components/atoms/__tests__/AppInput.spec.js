import {shallowMount} from '@vue/test-utils'
import AppInput from '@/components/atoms/AppInput.vue';

describe('AppInput.vue', () => {
    it('Should render AppInput', () => {
        const wrapper = shallowMount(AppInput, {
            props: {
                modelValue: 'some input value',
                placeholder: 'some placeholder',
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it('Should emit enter event', () => {
        const wrapper = shallowMount(AppInput, {
            props: {
                modelValue: 'some input value',
                placeholder: 'some placeholder',
            }
        });

        wrapper.find('input').trigger('keyup.enter');

        expect(wrapper.emitted()).toHaveProperty('enter');
    });

    it('Should emit esc event', () => {
        const wrapper = shallowMount(AppInput, {
            props: {
                modelValue: 'some input value',
                placeholder: 'some placeholder',
            }
        });

        wrapper.find('input').trigger('keyup.esc');

        expect(wrapper.emitted()).toHaveProperty('esc');
    });

    it('Should emit update:modelValue event with input value', () => {
        const wrapper = shallowMount(AppInput, {
            props: {
                modelValue: 'some input value',
                placeholder: 'some placeholder',
            }
        });

        wrapper.find('input').trigger('input');

        const updateModelValueEvent = wrapper.emitted('update:modelValue');

        expect(updateModelValueEvent[0]).toEqual(['some input value']);
    });
})
