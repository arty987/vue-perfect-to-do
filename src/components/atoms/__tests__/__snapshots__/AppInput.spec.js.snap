// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInput.vue Should render AppInput 1`] = `
<input
  class="app-input"
  maxlength="150"
  placeholder="some placeholder"
  type="text"
  value="some input value"
/>
`;
